### ISSUE TITLE
amp-carousel-0.2: Compilation error in dist build

### ISSUE THREAD

**Issue by choumx** [ampproject/amphtml#23966]
http://localhost:8000/examples/carousel.amp.html

```
v0.js:18 [Resource] failed to build: amp-carousel#17 ReferenceError: Carousel$$module$extensions$amp_base_carousel$0_1$carousel is not defined
    at R.push.f.k.buildCallback (amp-carousel-0.2.js:13)
    at v0.js:271
    at new Promise (<anonymous>)
    at HTMLElement.Jj.b.build (v0.js:270)
    at ej.f.build (v0.js:240)
    at mk (v0.js:323)
    at lk (v0.js:322)
    at ek.f.upgraded (v0.js:324)
    at HTMLElement.Jj.b.Ld (v0.js:269)
    at HTMLElement.Jj.b.Cg (v0.js:284)
```

**Comment by choumx** [comment]
Looks like `amp-base-carousel` isn't being bundled: https://github.com/ampproject/amphtml/blob/9d90a6addea6a405fb2b8cd6361e90a85d6c6936/extensions/amp-carousel/0.2/amp-carousel.js#L20

**Comment by jridgewell** [comment]
This should have caused a presubmit error. You're not supposed to import another extension's code (even another version of the same extension).

**Comment by sparhami** [comment]
> This should have caused a presubmit error. You're not supposed to import another extension's code (even another version of the same extension).

Yeah there is a whitelist, but I assumed that we could actually add to it since other extensions seem do it. Since there is no comments on why it exists, I assumed it was to prevent double inclusion of code (which I was okay with in this case) rather than the build failing to include the code.

**Comment by jridgewell** [comment]
In multipass builds, each extension is compiled individually with just `extensions/FOO/VERSION` and `src/` included for closure. If you import code like this, the closure invocation has to be updated as well.
